integrations/outlook/src/misc/custom-schemas.ts

```diff
@@ -0,0 +1,52 @@
+import * as z from "zod";
+
+export const notificationContentSchema = z
+  .object({
+    body: z.object({
+      content: z.string(),
+      contentType: z.string(),
+    }),
+    sender: z.object({
+      emailAddress: z.object({
+        address: z.string(),
+      }),
+    }),
+    from: z.object({
+      emailAddress: z.object({
+        address: z.string(),
+      }),
+    }),
+    toRecipients: z.array(
+      z.object({
+        emailAddress: z.object({
+          address: z.string(),
+        }),
+      })
+    ),
+    ccRecipients: z.array(
+      z.object({
+        emailAddress: z.object({
+          address: z.string(),
+        }),
+      })
+    ),
+    conversationId: z.string(),
+    subject: z.string(),
+    id: z.string(),
+  })
+  .superRefine((val, ctx) => {
+    if (val.body.content === "") {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: `Invalid email body: ${JSON.stringify(val, null, 3)}`,
+        path: ["body", "content"],
```

Comment:
try to avoid using refine, we convert these zod schemas to json schemas and then back to zod and these validations are lost. Instead you can use `message: z.string().min(1, 'email message must be defined')` (if the original zod error message is not clear enough)

Reply:
I agree here, this is a **chore**